python/python/tests/tests_server_modules/search/test_ft_search.py

```diff
@@ -0,0 +1,164 @@
+            index,
+            "*",
+            options=FtSeachOptions(
+                return_fields=[
+                    ReturnField(field_identifier="a", alias="a_new"),
+                    ReturnField(field_identifier="b", alias="b_new"),
+                ]
+            ),
+        )
+        # Check if we get the expected result from ft.search for string inputs
+        TestFtSearch._ft_search_deep_compare_result(
+            self,
+            result=result1,
+            json_key1=json_key1,
+            json_key2=json_key2,
+            json_value1=json_value1,
+            json_value2=json_value2,
+        )
+
+        # Search the index for byte inputs
+        result2 = await ft.search(
+            glide_client,
+            bytes(index, "utf-8"),
+            b"*",
+            options=FtSeachOptions(
+                return_fields=[
+                    ReturnField(field_identifier=b"a", alias=b"a_new"),
+                    ReturnField(field_identifier=b"b", alias=b"b_new"),
+                ]
+            ),
+        )
+
+        # Check if we get the expected result from ft.search from byte inputs
+        TestFtSearch._ft_search_deep_compare_result(
+            self,
+            result=result2,
+            json_key1=json_key1,
+            json_key2=json_key2,
+            json_value1=json_value1,
+            json_value2=json_value2,
+        )
+
+    def _ft_search_deep_compare_result(
+        self,
+        result: List[Union[int, Mapping[TEncodable, Mapping[TEncodable, TEncodable]]]],
+        json_key1: str,
+        json_key2: str,
+        json_value1: dict,
+        json_value2: dict,
+    ):
+        type_name_bytes = "bytes"
+        assert len(result) == 2
+        assert result[0] == 2
+        searchResultMap: Mapping[TEncodable, Mapping[TEncodable, TEncodable]] = cast(
+            Mapping[TEncodable, Mapping[TEncodable, TEncodable]], result[1]
+        )
+        for key, fieldsMap in searchResultMap.items():
+            keyString = key
+            if type(key).__name__ == type_name_bytes:
+                print(type(key).__name__)
+                keyString = cast(bytes, key).decode(encoding="utf-8")
+            assert keyString == json_key1 or keyString == json_key2
+            if keyString == json_key1:
+                for fieldName, fieldValue in fieldsMap.items():
+                    fieldNameString = fieldName
+                    if type(fieldName).__name__ == type_name_bytes:
+                        fieldNameString = cast(bytes, fieldName).decode(
+                            encoding="utf-8"
+                        )
+                    fieldValueInt = int(fieldValue)
+                    if type(fieldValue).__name__ == type_name_bytes:
+                        fieldValueInt = int(
+                            cast(bytes, fieldValue).decode(encoding="utf-8")
+                        )
+                    assert fieldNameString == "a" or fieldNameString == "b"
```

Comment:
what is that? str = a or b?
Create expected result structure and compare with it

Reply:
yes

Reply:
yes, str == a or str ==b

Reply:
When you change input data for tests, you'll have to update validation in a lot of places.
Don't do that, create expected result instead and compare with it.

Reply:
Updated.